src/main/scala/subsystem/BaseSubsystem.scala

```diff
@@ -74,7 +74,27 @@ abstract class BaseSubsystem(implicit p: Parameters) extends BareSubsystem with
     }
   }
 
-  val logicalTreeNode = new SubSystemLogicalTreeNode()
+  def getCoreComplexRTLModule(): Option[OMCoreComplexRTLModule] = {
+    val rtlIfc = CoreComplexRTLInterface(
+      clocks = Nil,
+      clockRelationships = Nil,
+      resets = Nil,
+      statuses = Nil,
+      localInterrupts = None, // E.g. local_interrupts_X (all local interrupts for core X)
+      globalInterrupts = None, // E.g. global_interrupts
+      machineExternalInterrupts = None, // E.g. meip_X
+      testModeSignals = Nil // E.g. debug_psd_test_mode and debug_psd_test_mode_reset
+    )
```

Comment:
What is the purpose of filling in all of these fields with `None` and `Nil`? If the whole function returns an `Option[OMCoreComplexRTLModule]`, why not just return `None` for the whole thing instead of filling out null values for each individual field?

Reply:
I'm assuming this is where designer should fill in their contents?

Reply:
But there's no purpose to this object instance; it doesn't represent anything that couldn't be represented with a `None`. If someone else wanted to later fill in a value, they could easily just pattern match on the `None` and fill in their own values in the same way.

Reply:
Ok, I think I understand what you're saying... let me rephrase it here and you sanity check?

It's basically expected that someone is going to override this `  def getCoreComplexRTLModule(): Option[OMCoreComplexRTLModule] = ` entirely. And therefore, there is no reason that this default version shouldn't just return `None`, vs bothering to fill out a bunch of useless instances.

is that what you mean?

Reply:
Yes, that's what I mean, although I don't know exactly what the original intent of this function was.

Reply:
I assumed that Derek was just creating the skeleton, then would ask designer to come modify *this code* to fill in the appropriate values, as part of this PR. At any rate I will attempt to do *that* to see if it makes sense, otherwise I'll just return `None` as you explained.